File: 力扣/0-1/3_找出字符串中第一个匹配项的下标/main.js
function strStr(haystack, needle) {
    if (needle.length === 0) {
        return 0;
    }
    
    for (let i = 0; i <= haystack.length - needle.length; i++) {
        if (haystack.substring(i, i + needle.length) === needle) {
            return i;
        }
    }
    return -1;
}

console.log(strStr("sadbutsad", "sad"));    // 0
console.log(strStr("leetcode", "leeto"));   // -1

File: 力扣/0-1/3_找出字符串中第一个匹配项的下标/main.go
package main

import "fmt"

func strStr(haystack string, needle string) int {
	if len(needle) == 0 {
		return 0
	}

	for i := 0; i <= len(haystack)-len(needle); i++ {
		if haystack[i:i+len(needle)] == needle {
			return i
		}
	}
	return -1
}

func main() {
	fmt.Println(strStr("sadbutsad", "sad"))  // 0
	fmt.Println(strStr("leetcode", "leeto")) // -1
}

// sadbutsad

// sad

// 6

// haystack[i:i+len(needle)] 切片操作
// i=0 [0,3)
